src/expr/src/scalar.rs

```diff
@@ -1027,6 +1028,24 @@ impl MirScalarExpr {
                                     e.typ(column_types).scalar_type,
                                 ),
                             }
+                        } else if *func == BinaryFunc::ToCharTimestamp && expr2.is_literal() {
+                            let format_str = expr2.as_literal_str().unwrap();
+                            *e = MirScalarExpr::CallUnary {
+                                func: UnaryFunc::ToCharTimestamp(func::ToCharTimestamp {
+                                    format_string: format_str.to_string(),
+                                    format: DateTimeFormat::compile(format_str),
+                                }),
+                                expr: Box::new(expr1.take()),
+                            };
+                        } else if *func == BinaryFunc::ToCharTimestampTz && expr2.is_literal() {
```

Comment:
Can these two `if` blocks be folded into one and do the conditional when constructing `func` instead?

Reply:
Yeah it's a trade-off. It just bugged me a little that most of the block is the same except the new `func`. But it's fine to duplicate it for now to keep it consistent with the existing pattern, as you mentioned.

Just for reference, I had the following in mind:
```rust
else if matches!(
    *func,
    BinaryFunc::ToCharTimestamp | BinaryFunc::ToCharTimestampTz
) && expr2.is_literal()
{
    let fmt_str = expr2.as_literal_str().unwrap();
    let fmt = func::format::DateTimeFormat::compile(&fmt_str);
    *e = MirScalarExpr::CallUnary {
        func: if *func == BinaryFunc::ToCharTimestamp {
            UnaryFunc::ToCharTimestamp(fmt)
        } else {
            UnaryFunc::ToCharTimestampTz(fmt)
        },
        expr: Box::new(expr2.take()),
    };
}
```